### ISSUE TITLE
[Bug]: "test_graph_info_load_from_s3" failed under build type of Release

### ISSUE THREAD

**Issue by lixueclaire** [alibaba/GraphAr#184]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When I use `cmake -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTS=ON ..` and then `make test`, I got an error:

> Start  6: test_graph_info_load_from_s3
> Test   6: test_graph_info_load_from_s3 ................***Exception: SegFault 10.66 sec

### Expected Behavior

pass this test

### Minimal Reproducible Example

```
cmake -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTS=ON ..
make test
```

### Environment

- Operating system: macOs Ventura 13.3.1
- GraphAr version: v0.5.0

**Comment by lixueclaire** [comment]
@acezen could you please help to have a look at this issue?

**Comment by acezen** [comment]
> @acezen could you please help to have a look at this issue?

Ok, I'll take this and try to reproduce the bug.

**Comment by acezen** [comment]
seems it's arrow 12.0.0 bring the bug
https://github.com/apache/arrow/blob/fab79f50caf6a80eccaa3bbae79d518af7b408e5/cpp/src/arrow/filesystem/s3fs.cc#L2597-L2603

**Comment by acezen** [comment]
> 

En... It seems than arrow bring a `AwsInstance` to manager the S3 filesystem in all arrow threads BUT arrow did not take care the life cycle of AwsInstance, they give the life cycle control to user, and user need to call `EnsureS3Finalized` when the S3 job is done

**Comment by acezen** [comment]
Consider to make patch to GraphAr that add `FinalizedS3` static function to `FileSystem` and let user to finalize the S3 system handly.

**Comment by lixueclaire** [comment]
> Consider to make a patch to GraphAr that add a `FinalizedS3` static function to `FileSystem` and let user to finalize the S3 system manually. @lixueclaire what do you think?

I agree that it would be helpful to add such a function to FileSystem in GraphAr to allow users to finalize the S3 system manually. And we could document this feature in the documentation to ensure that users are aware of this special processing requirement for the S3 system.

**Comment by acezen** [comment]
> 

agree, thanks for the advise.

**Comment by acezen** [comment]
Open a keep track of upstream fix issue #187 ,close this one.
